toc.md

```diff
@@ -165,34 +190,50 @@ module Workerholic
 end
 ```
 
-Our internal `kill` method simply sets the `alive` instance variable from `true` to `false`. Afterwards, we join each of our worker threads with the main thread. This way, we ensure that the workers can have a chance to finish their final jobs before actually shutting down.
+This will allow the worker to finish executing its current job. Afterwards, we `join` each of our worker with our main thread of execution. We will touch on why this is important in a later section, for now it is only useful to know that the workers will be waited on before Workerholic exits.
 
 ### Efficiency
-Let's shift gears a little bit and start talking about how Workerholic is efficient.
+
+Efficiency is another core feature for a BJP. Why does a BJP need to be efficient? How can we build an efficient? Those are the questions that we will answer in the following sections.
 
 #### An Example Scenario
 
 ##### Scenario
-Suppose we have a huge Rails application with the follow statistics:
+
+Suppose we have a significantly big Rails application with the following statistics:
 * 1000 average queries per second (QPS)
-* 10% of those is email sending
-* 1% image processing
+* 10% of these queries involve some email sending
+* 1% of these queries involve some image processing
 
-Also suppose that we a machine with the follow specs:
+Also suppose we have a machine with the following specs:
 * 4GB RAM
-* 4 CPU cores.
+* 4 CPU cores
 
 ##### Challenge
-The challenge we have here is how do we maximize the use of available resources?
+
+How can we build a BJP that will maximize the use of available hardware resources?
 
 #### Concurrency
 
 ##### Email Jobs calculations
-To begin braeking down our example scenario, let's make the assumption that it would take an average of 50ms to send an email; this is the time it takes between making the request and getting a response from the email service. If we take a 24 hour window, at 86400 seconds per day and 1000 QPS, we will get 86.4 million requests per day, 10% of which are email which means 8.64 million email sending background jobs. If each takes an average of 50ms, it means that in a span of 24 hours, we have a list of jobs that will take 120 hours to process, which gives us an enqueuing:processing ratio of 1:5. If we left this unchecked and allowed these to be performed synchronously, then over longer periods of time, we will get a backlog of background jobs.
 
-![serialized_job_length_redis](/images/serialized_job_length_redis.png)
+First of all we need to do a bit of calculations in order to understand the problem we are facing with the current scenario we have:
```

Comment:
First of all we need to do a bit of calculations in order to understand the problem we face in this particular scenario: